src/main/java/com/ericsson/ei/subscriptionhandler/InformSubscriber.java

```diff
@@ -165,21 +155,102 @@ public void informSubscriber(String aggregatedObject, JsonNode subscriptionJson)
                 LOGGER.error(e.getMessage());
             }
         }
-        // set as empty because we should avoid leaking of authentication details
-        key = "";
-        val = "";
+
+    }
+
+    /**
+     * This method prepares headers to be used when making a rest call with the
+     * method POST.
+     *
+     * @param headers
+     * @param notificationMeta
+     * @param subscriptionJson
+     * @return
+     */
+    private HttpHeaders prepareHeaders(HttpHeaders headers, String notificationMeta, JsonNode subscriptionJson) {
+        // Setting Content Type
+        String headerContentMediaType = getSubscriptionField("restPostBodyMediaType", subscriptionJson);
+        headers.setContentType(MediaType.valueOf(headerContentMediaType));
+        LOGGER.debug("Successfully added header: "
+                + String.format("'%s':'%s'", "restPostBodyMediaType", headerContentMediaType));
+
+        // Adding authentication if any
+        String authType = getSubscriptionField("authenticationType", subscriptionJson);
+        if (authType.equals("BASIC_AUTH")) {
+            String username = getSubscriptionField("userName", subscriptionJson);
+            String password = getSubscriptionField("password", subscriptionJson);
+
+            if (!username.equals("") && !password.equals("")) {
+                String encoding = Base64.getEncoder().encodeToString((username + ":" + password).getBytes());
+                headers.add("Authorization", "Basic " + encoding);
+                LOGGER.debug("Successfully added header for 'Authorization'");
+
+                // Adding jenkins crumb if any
+                JsonNode JenkinsJsonCrumbData = fetchJenkinsCrumbIfAny(encoding, notificationMeta);
```

Comment:
make this section to its own method. Variable name starts with lower letter.

Reply:
Done